projectq/backends/_awsbraket/_awsbraket.py

```diff
@@ -0,0 +1,481 @@
+            self._circuit += '{"control": ' + str(ctrl_pos) + \
+                             ', "target": ' + str(qb_pos) + \
+                             ', "type": "c' + gatetxt + '"}, '
+            return
+        if gate == X and get_control_count(cmd) == 1:
+            ctrl_pos = cmd.control_qubits[0].id
+            qb_pos = cmd.qubits[0][0].id
+            self._circuit += '{"control": ' + str(ctrl_pos) + \
+                             ', "target": ' + str(qb_pos) + \
+                             ', "type": "cnot"}, '
+            return
+        if gate == Swap and get_control_count(cmd) == 1:
+            ctrl_pos = cmd.control_qubits[0].id
+            qb_pos0 = cmd.qubits[0][0].id
+            qb_pos1 = cmd.qubits[1][0].id
+            self._circuit += '{"targets": [' + \
+                str(qb_pos0) + ', ' + str(qp_pos1) + \
+                ' ], \
+                "control": ' + str(ctrl_pos) + \
+                '"type": "cswap"}, '
+            return
+        if gate == X and get_control_count(cmd) == 2:
+            ctrl_pos0 = cmd.control_qubits[0].id
+            ctrl_pos1 = cmd.control_qubits[1].id
+            qb_pos = cmd.qubits[0][0].id
+            self._circuit += '{"controls": [ ' + \
+                             str(ctrl_pos0) + ', ' + str(ctrl_pos1) + \
+                             ' ], \
+                                "target": ' + str(qb_pos) + \
+                             ', "type": "ccnot"}, '
+            return
+        if get_control_count(cmd) == 0:
+            if isinstance(gate, (R, Rx, Ry, Rz)):
+                qb_pos = cmd.qubits[0][0].id
+                angle = gate.angle
+                gatetxt = self._gationary[gate.__str__().split('(')[0]]
+                self._circuit += '{"angle": ' + str(angle) + \
+                                 ', "target": ' + str(qb_pos) + \
+                                 ', "type": "' + gatetxt + '"}, '
+                return
+            if gate in (X, Y, Z, H, S, T, Sdag, Tdag, SqrtX):
+                qb_pos = cmd.qubits[0][0].id
+                gatetxt = self._gationary[gate.__str__()]
+                self._circuit += '{"target": ' + str(qb_pos) + \
+                                 ', "type": "' + gatetxt + '"}, '
+                return
+            if gate == Swap:
+                qb_pos0 = cmd.qubits[0][0].id
+                qb_pos1 = cmd.qubits[1][0].id
+                self._circuit += '{"targets": [' + \
+                                 str(qb_pos0) + ', ' + str(qp_pos1) + \
+                                 '], \
+                                 "type": "swap"}, '
+                return
+            # TODO: Add unitary for the SV1 simulator as MatrixGate
+        if gate == Barrier:
+            return
+        raise Exception('Invalid command: ' + str(cmd) +
+                        '. Please check the available commands'
+                        'for the device ' +
+                        self.device + '.')
```

Comment:
I have restructured the code following your suggestions. After thinking for a while I did not created the helper function to create the circuit pieces of json because I think it will just repeat the different cases structure due to the angles, controls and targets. Please hace a look

Reply:
In addition, take into account the previous open comment on the use or not of a helper function

Reply:
After thinking about it some more (see my other [comment](https://github.com/ProjectQ-Framework/ProjectQ/pull/388/files?file-filters%5B%5D=.py&file-filters%5B%5D=.rst#r599678989)), a general approach is indeed possible and does not require what I was suggesting here.